### PR TITLE
Windows build

### PR DESCRIPTION
**Pull request by supunkamburugamuve**

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        else:
+            print("The build should be in a conda environment")
+            return
+
+    cmake_command = f'cmake -DPYCYLON_BUILD={on_off(BUILD_PYTHON)} -A x64 \
+      -DCMAKE_BUILD_TYPE={CPP_BUILD_MODE} -DCYLON_WITH_TEST={on_off(RUN_CPP_TESTS)} -DARROW_BUILD_TYPE=SYSTEM {CPPLINT_COMMAND} {INSTALL_COMMAND} \
+      {CMAKE_FLAGS} {CPP_SOURCE_DIR}'
+
+    print(cmake_command, BUILD_DIR)
+    res = subprocess.call(cmake_command, cwd=BUILD_DIR, shell=True)
```
Comment (nirandaperera) [inline build.py:136]: don't we have to forward the environment variables to the subprocess here? expecially in the conda environment?

File: cpp/CMake/Modules/ConfigureArrow.cmake [hunk 2]
```diff
@@ -56,9 +56,15 @@ set(ARROW_CMAKE_ARGS " -DARROW_WITH_LZ4=OFF"
 if (PYCYLON_BUILD)
     find_package(Python3 COMPONENTS Interpreter REQUIRED)
     message("Python Executable Path ${Python3_EXECUTABLE}")
-
-    list(APPEND ARROW_CMAKE_ARGS " -DARROW_PYTHON=${PYCYLON_BUILD}"
-            " -DPYTHON_EXECUTABLE=${Python3_EXECUTABLE}")
+    if(WIN32)
+        list(APPEND ARROW_CMAKE_ARGS " -DARROW_PYTHON=${PYCYLON_BUILD}"
+                " -DPYTHON_EXECUTABLE=${Python3_EXECUTABLE}"
+                " -DBOOST_ROOT=C:/local/boost_1_77_0_b1_rc1")
```
Comment (nirandaperera) [inline cpp/CMake/Modules/ConfigureArrow.cmake:62]: should we leave this hard-coded?

File: cpp/CMake/Templates/Arrow.CMakeLists.txt.cmake [hunk 3]
```diff
@@ -16,6 +16,8 @@ cmake_minimum_required(VERSION 3.10)
 project(ArrowModule)
 include(ExternalProject)
 
+
+message("############# ${CYLON_ARROW_VERSION}")
```
Comment (nirandaperera) [inline cpp/CMake/Templates/Arrow.CMakeLists.txt.cmake:20]:
```suggestion

```
lets remove this

File: cpp/test/CMakeLists.txt [hunk 4]
```diff
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-SET(MPI_RUN_CMD mpirun)
+SET(MPI_RUN_CMD mpiexec)
```
Comment (nirandaperera) [inline cpp/test/CMakeLists.txt:18]:
```suggestion
find_package(mpi REQUIRED)
```

File: cpp/test/CMakeLists.txt [hunk 5]
```diff
@@ -43,7 +43,11 @@ endfunction(cylon_add_test)
 # macro to run executable
 function(cylon_run_test TESTNAME no_mpi_proc)
   set(exec_name "${TESTNAME}_${no_mpi_proc}")
-  set(test_params --allow-run-as-root --oversubscribe -np ${no_mpi_proc} "${CMAKE_BINARY_DIR}/bin/${TESTNAME}")
+  if (WIN32)
+    set(test_params -np ${no_mpi_proc} "${CMAKE_BINARY_DIR}/bin/${CMAKE_BUILD_TYPE}/${TESTNAME}")
+  else()
+    set(test_params --allow-run-as-root --oversubscribe -np ${no_mpi_proc} "${CMAKE_BINARY_DIR}/bin/${TESTNAME}")
+  endif()
   add_test(NAME ${exec_name} COMMAND ${MPI_RUN_CMD} ${test_params})
```
Comment (nirandaperera) [inline cpp/test/CMakeLists.txt:51]:
```suggestion
  add_test(NAME ${exec_name} COMMAND ${MPIEXEC_EXECUTABLE} ${test_params})
```

**Review by nirandaperera — APPROVED** [review]
I made some minor comments

File: build.py (hunk 1 above)
Comment (supunkamburugamuve) [reply build.py:136]: I think we don't need to.

File: cpp/CMake/Modules/ConfigureArrow.cmake (hunk 2 above)
Comment (supunkamburugamuve) [reply cpp/CMake/Modules/ConfigureArrow.cmake]: Removed this

**Pull request merged**